pyproject.toml

```diff
@@ -46,17 +46,10 @@ pytest-rerunfailures = "^11.1.2"
 coverage = "^7.2.6"
 
 [tool.poetry.group.lint.dependencies]
+ruff = "^0.0.278"
```

Comment:
I assume that is the intent behind having the ^ here. Most of the other dev tools have this specification in pyproject.toml.

Reply:
> All it's likely to do in the updates is adding new rules / fix bugs on existing rues.

Exactly this can fail our CI, the same way as pyright sometimes fails after update.

> for which we also use a carrot dependency

We actually shouldn't.

> thing is, these tools can't just count every change to a rule as a breaking change, or their major version would be really high.

Mypy somehow have their [sem-ver like versioning](https://mypy-lang.blogspot.com/2023/02/mypy-10-released.html), that resolves this problem.

Reply:
I'm okay with updates potentially failing in CI. They are handled individually by Dependabot and allow us to preview upcoming changes. Those results enable us to report bugs if necessary. Or, by waiting until the next update PR, the problem may disappear without any intervention.

Reply:
Okay, never mind, I don't really want to block this PR because of one symbol removing.

Reply:
This can always be done later, in another PR which perhaps does the same for pyright and any other deps that need it. I'm still not sure we should be doing this though, so if you will make another PR for it, do add a good description explaining why it should be done.

Reply:
`poetry update` is a tiny part of the development workflow, and Dependabot mainly handles it. With the locked version already passing CI tests, contributors are guaranteed to have a working version unless they do Dependabot's job. If a contributor messes with package versions, some code likely needs to be updated in mcstatus or a bug in a package they are investigating.